lib/bloc/podcast_clip/podcast_clip_bloc.dart

```diff
@@ -0,0 +1,206 @@
+import 'dart:async';
+import 'dart:io';
+import 'dart:typed_data';
+import 'package:anytime/services/audio/audio_player_service.dart';
+import 'package:breez/bloc/async_actions_handler.dart';
+import 'package:breez/bloc/podcast_clip/podcast_clip_details_model.dart';
+import 'package:ffmpeg_kit_flutter_https_gpl/ffmpeg_kit.dart';
+import 'package:ffmpeg_kit_flutter_https_gpl/ffmpeg_session.dart';
+import 'package:ffmpeg_kit_flutter_https_gpl/return_code.dart';
+import 'package:path_provider/path_provider.dart';
+import 'package:rxdart/rxdart.dart';
+
+const int _initialSeconds = 10;
+const int _maxClipDuration = 120;
+
+class PodcastClipBloc with AsyncActionsHandler {
+  final BehaviorSubject<PodcastClipDetailsModel> _clipDetailsBehaviourSubject =
+      BehaviorSubject();
+
+  Stream<PodcastClipDetailsModel> get clipDetails =>
+      _clipDetailsBehaviourSubject.stream;
+
+  setPodcastClipDetails({PositionState position}) {
+    Duration endTime = position.position + Duration(seconds: _initialSeconds);
+
+    PodcastClipDetailsModel podcastClipDetails = PodcastClipDetailsModel(
+        startTimeStamp: position.position,
+        endTimeStamp: endTime,
+        clipDuration: _initialSeconds,
+        episodeLength: position.length,
+        episodeDetails: position.episode,
+        podcastClipState: PodcastClipState.IDLE);
+
+    _clipDetailsBehaviourSubject.add(podcastClipDetails);
+  }
+
+  setPodcastClipDuration({int durationInSeconds}) async {
+    var clipDetails = getCurrentPodcastClipDetails();
+
+// Check if the duration exceeds the episode length
+    if ((clipDetails.startTimeStamp + Duration(seconds: durationInSeconds)) >
+        clipDetails.episodeLength) {
+      return;
+    }
+
+    Duration clipEndTime =
+        clipDetails.startTimeStamp + Duration(seconds: durationInSeconds);
+
+    _clipDetailsBehaviourSubject.add(clipDetails.copy(
+        clipDuration: durationInSeconds, endTimeStamp: clipEndTime));
+  }
+
+  incrementDuration() async {
+    var clipDetails = getCurrentPodcastClipDetails();
+    int currentClipDuration = clipDetails.clipDuration;
+
+    //If duration is less than 10 seconds return
+    if (_maxClipDuration - currentClipDuration <= 0) {
+      return;
+    }
+
+    int incrementedClipDuration = 0;
+    if (currentClipDuration % 10 != 0) {
+      incrementedClipDuration =
+          currentClipDuration - currentClipDuration % 10 + _initialSeconds;
+    } else {
+      incrementedClipDuration = currentClipDuration + _initialSeconds;
+    }
+    setPodcastClipDuration(durationInSeconds: incrementedClipDuration);
+  }
+
+  decrementDuration() async {
+    var clipDetails = getCurrentPodcastClipDetails();
+    int currentClipDuration = clipDetails.clipDuration;
+
+    //return if duration needs to be greator than 10 seconds
+    if (currentClipDuration <= _initialSeconds) {
```

Comment:
Seems like ensuring the clip duration is within bounds belongs in the setPodcastClipDuration function.

Reply:
> @utkarshshendge this is a good work! thanks for addressing the feedback quickly. I went over the PR again and added my final feedback. Thank you.

@roeierez I have done the changes. Thanks